src/compiler/binder.ts

```diff
@@ -2563,7 +2563,7 @@ function createBinder(): (file: SourceFile, options: CompilerOptions) => void {
     }
 
     function checkStrictModeFunctionName(node: FunctionLikeDeclaration) {
-        if (inStrictMode) {
+        if (inStrictMode && !(node.flags & NodeFlags.Ambient)) {

```

Comment:
This will allow for un-namespaced `declare function eval(): void` too, right? Is there any legal strict-mode code that could be typed by that? I can't think of any.

Restricting this exemption to namespaces makes sense to me if there aren't any other examples with legal emit.

Reply:
> This will allow for un-namespaced declare function eval(): void too, right?

Yes, I've added tests to cover this case

Reply:
I'd like to disallow `declare function eval(): void`. You can't write a matching `function eval() { }` or `var eval = function() { }`, can you?

Reply:
How do _we_ declare the global `eval` if that's illegal?

Reply:
good point!